### DISCUSSION TITLE
`useSWRMutation`: Mutate without `key` prop?

### DISCUSSION THREAD

**Discussion by psteinroe** [vercel/swr#2461] [Q&A] [answered]
Hey,

I am currently evaluating whether to replace [`use-mutation`](https://www.npmjs.com/package/use-mutation) with the new `useSWRMutation` hook. I have a rather complex use case, where I revalidate multiple keys after a successful mutation. Some keys are just revalidated, and for some the cache is populated. With `use-mutation`, I simply handle the cache update within the `onSuccess` callback. A similar API is provided by `tanstack-query`. I am trying to understand if that is something that the new `useSWRMutation` hook supports too. It seems like the mutation is always bound to a single key (filter functions are not supported either). I also need the key of the current cache item within  `populateCache`, and execute different branches based on it. 

Best
Philipp

**Comment by shuding** [comment, marked as answer]
Hi! With the key provided to `useSWRMutation`, SWR can be aware of which resource the mutation might affect. This can help handle race conditions and revalidation correctly together with `useSWR`.

For example, when a fetch happens via `useSWR` when a mutation for the same resource is still ongoing, SWR can discard the fetch result and re-do it after the mutation finishes:

```js
const { data } = useSWR('/api/user', fetcher)
const { trigger } = useSWRMutation('/api/user', changeName)
```

But, if you want to use `useSWRMutation` without strictly binding to a specific resource, you can give it a random key, such as:

```js
const randomKey = useState(() => Math.random())[0]
const { ... } = useSWRMutation(randomKey, ...)
```

You can pack that as a custom hook too.

**Reply by benevbright** [reply]
> you can give it a random key, such as:

@shuding 
I can also use a unique string for that mutation, right? 

just like `"nokey-updateXX"`. Is there a side effect? Ideally I would like to put `null` in that case.
